infra/asttools.py

```diff
@@ -113,9 +120,17 @@ def visit_ImportFrom(self, cmd):
 class ResolvePatches(ast.NodeTransformer):
     def __init__(self):
         self.patches = {}
+        self.patchables = {}
 
     def visit_FunctionDef(self, cmd):
-        if not cmd.decorator_list:
+        if not cmd.decorator_list or not is_patch_decorator(cmd.decorator_list[0]):
+            if cmd.decorator_list:
+                if not is_patchable_decorator(cmd.decorator_list[0]):
+                    print(cmd.decorator_list)
+                    print(isinstance(cmd, ast.Call))
+                    raise Exception()
```

Comment:
I assume this is debugging code. IIRC there's some mechanism in the compiler for nice error messages, don't remember for sure though.